base_layer/core/src/base_node/state_machine.rs

```diff
@@ -66,6 +66,7 @@ pub struct BaseNodeStateMachine<B: BlockchainBackend> {
     pub(super) connection_manager: ConnectionManagerRequester,
     pub(super) metadata_event_stream: Subscriber<ChainMetadataEvent>,
     pub(super) config: BaseNodeStateMachineConfig,
+    pub(super) info: StateInfo,
```

Comment:
I would create a new PubSub pair and let clients subscribe to it:

```rust
   ...
    event_sender: Publisher<StateEvent>,
    event_receiver: Subscriber<StateEvent>,
    interrupt_signal: ShutdownSignal,
    status_event_publisher: Publisher<StatusInfo>,   // NEW
    status_event_subscriber: Subscriber<StatusInfo>,// NEW
}

impl<B: BlockchainBackend + 'static> BaseNodeStateMachine<B> {
    /// Instantiate a new Base Node.
    pub fn new(...) -> Self
    {
        let (event_sender, event_receiver): (Publisher<_>, Subscriber<_>) = bounded(10);
        let (status_event_publisher, status_event_subscriber): (Publisher<_>, Subscriber<_>) = bounded(10);// NEW
        Self {
            ...
            event_sender,
            event_receiver,
            status_event_publisher,// NEW
            status_event_subscriber,// NEW
        }
    }


...


    pub fn get_status_event_stream(&self) -> Subscriber<StateEvent> {
        self.status_event_subscriber.clone()
    }
```

Reply:
done